client/src/pages/inventory/VariantCreate.tsx

```diff
@@ -140,10 +140,13 @@ const VariantCreateDialog: React.FC<VariantCreateDialogProps> = ({
         <LoadingButton
           loading={isCreateVariantLoading}
           onClick={handleSubmit((data) => {
-            handleCreate({
-              ...data,
-              price: floatToPennies(data.price),
-            });
+            handleCreate(
+              {
+                ...data,
+                price: floatToPennies(data.price),
+              },
+              (variant as ßwillFixThisTypeLater)?.id ?? undefined,
```

Comment:
Can we type this now? I can see that the variant can either be `Variant` or `PostVariantBody`, the latter of which does not have id. Just wanted to ask how we could tackle this.

Reply:
We need to look at why there is a delta in the component, why is one component handling the duty of two vastly different flows. This requires a larger scoped rework than we might anticipate so I say this is something we should dedicate some time to in the future.

The page was not well implemented before actually, it didn't even have variant PUT integrated, so whenever you update a variant it just POSTs a new one, I just got it to work for now but if we ever get time and evolve our product we should tackle it.